configman/config_manager.py

```diff
@@ -556,40 +556,45 @@ def get_option_names(self):
                 if isinstance(self.option_definitions[x], Option)]
 
     #--------------------------------------------------------------------------
-    def _create_reference_value_from_links(self, keys, known_keys):
+    def _create_reference_value_options(self, keys, finished_keys):
         """this method steps through the option definitions looking for
         alt paths.  On finding one, it creates the 'reference_value_from' links
         within the option definitions and populates it with copied options."""
         # a set of known reference_value_from_links
-        set_of_reference_value_from_links = set()
-        for key in (k for k in keys if k not in known_keys):
+        set_of_reference_value_option_names = set()
```

Comment:
cool. I have no problem with that. 
As long as you don't adopt the old VB style of strFirstname (for strings) or fooInt (for integers) :)

I'm big on PEP8 but even bigger on "Pythonic'ism" and what comes with that is that you avoid saying yourself into a corner. E.g. instead of calling it "list_of_names" you call it "iterable" or "sequence" and then functions can become generic. E.g. 

```python
def check_empties(iterable):
    for thing in iterable:
        if not thing:
            raise MyException(thing)
```
Code like this would be impossible with statically typed languages because you'd have to write (in Go for example):
```go
func check_empties(iterable []string):
    for _, thing := range iterable {
       if thing == "" {
          panic(thing)
       }
    }
```
which is tied to list of strings only. The pythonic version doesn't care what the type is as long as the thing has an `__iter__` and that each element in the sequence has a `__bool__` (or whatever the boolean operator is called). 

I'm not defending anything or trying to make you change. Never! But I think it's worth backing up my comment with something that just isn't a "gut feeling".